### PR TITLE
Added Solution for NQueens Problem no 51 from Leetcode

### PR DESCRIPTION
**Pull request by fame2105**
Problem Description :
https://leetcode.com/problems/n-queens/

### CODE REVIEW

File: 3]. Competitive Programming/LeetCode/NQueens_51.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
+
+class NQueens_51 {
+    public static List<List<String>> solveNQueens(int n) {
+        // create n x n Board
+        int[][] board = new int[n][n];
+        List<List<String>> result = new ArrayList<>();
+        solveNQueensHelper(board, 0, new ArrayList<String>(), result);
+        return result;
```
Comment (AkashSingh3031) [inline 3]. Competitive Programming/LeetCode/NQueens_51.java:8]: Error
```suggestion
        return result;
```

**Review by AkashSingh3031 — CHANGES_REQUESTED** [review]
Please Check your solution again

And create folder "Problems" and solution code
like   '_0051)_N-Queens.java'

**Comment by AkashSingh3031** [comment]
@fame2105 Please check your solution and send a pull request again

File: 3]. Competitive Programming/LeetCode/NQueens_51.java (hunk 1 above)
Comment (fame2105) [reply 3]. Competitive Programming/LeetCode/NQueens_51.java]: Code is error free. Infact, this is the solution that I was able to submit on Leetcode successfully.
Could you please elaborate on the error that you got when you tried this solution ?

Comment (AkashSingh3031) [reply 3]. Competitive Programming/LeetCode/NQueens_51.java]: image

check this

Comment (AkashSingh3031) [reply 3]. Competitive Programming/LeetCode/NQueens_51.java]: image

Change the class name in code of 1st line

And add the solution in folder ( `Problems` )
code name like `_0051)_N-Queens.java`

File: 3]. Competitive Programming/LeetCode/Problems/_0051)_N-Queens.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// Link to Problem : https://leetcode.com/problems/n-queens/
+class NQueens_51 {
```
Comment (AkashSingh3031) [inline 3]. Competitive Programming/LeetCode/Problems/_0051)_N-Queens.java:2]: change name `NQueens_51` to `Solution`
then code accepted

**Review by AkashSingh3031 — CHANGES_REQUESTED** [review]

File: 3]. Competitive Programming/LeetCode/NQueens_51.java (hunk 1 above)
Comment (fame2105) [reply 3]. Competitive Programming/LeetCode/NQueens_51.java]: Yes that makes absolute sense, solution provided in the file has been named as NQueens_51.
All the solutions on leetcode should be present in the Solution class, when you tried to run execute the solution, you have copied the entire content of the file.
Only copy the methods inside the provided class, and make sure that you do not copy the main method.

Sample Reference: 
Accepted Solution

File: 3]. Competitive Programming/LeetCode/Problems/_0051)_N-Queens.java (hunk 2 above)
Comment (fame2105) [reply 3]. Competitive Programming/LeetCode/Problems/_0051)_N-Queens.java]: Done. 👍

**Review by AkashSingh3031 — COMMENTED** [review]
Accepted

**Pull request merged**